app/src/main/java/com/google/droidjump/GameView.java

```diff
@@ -71,18 +75,32 @@ public void run() {
 
     private void receiveLevelDetails() {
         // TODO: Serialize current level data and put it in some container
+
         levelTimePoints = 200;
-        levelSpeed = 50;
+        levelSpeed = levelData.getBaseSpeed();
     }
 
     public void updateGameState() {
         // TODO: Check if time point is in level data and add data to some container, than move
         //  it to left
+        checkTimePoint();
         updateDroidCoordinates();
+    }
+
+    private void checkTimePoint() {
+        if (levelData.isEmpty()) {
 
-        // Level Finishing
-        if (timePoint == levelTimePoints) {
+            // When the obstacles end - the level is considered passed
             winGame();
+        } else if (timePoint == levelData.getCurTimeInterval()) {
+
+            //  This is just an example of how we can get
+            //  info about an obstacle that should appear at the moment
+
+            System.out.println("\n" + timePoint + " time to add new obstacle:");
+            String newObstacleType = levelData.getNewObstacleType();
+            System.out.println(newObstacleType + "\n");
+            timePoint = 0;
```

Comment:
This is a bad thing to do as the game won't finish when it is intended. "timePoint" represents the amount of time passed since the beginning of level while what you need is the amount of time passed since the last obstacle. Please add another variable for that.

Reply:
Done